src/openms/include/OpenMS/ANALYSIS/ID/ConsensusMapMergerAlgorithm.h

```diff
@@ -0,0 +1,151 @@
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Julianus Pfeuffer $
+// $Authors: Julianus Pfeuffer $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/DATASTRUCTURES/DefaultParamHandler.h>
+#include <OpenMS/CONCEPT/Exception.h>
+#include <OpenMS/CONCEPT/ProgressLogger.h>
+#include <OpenMS/KERNEL/ConsensusMap.h>
+#include <OpenMS/METADATA/ExperimentalDesign.h>
+
+#include <unordered_set>
+
+namespace OpenMS
+{
+  /**
+   * @brief Merges identification data in ConsensusMaps
+   * Has some stuff in common with IDMergerAlgorithm and therefore could be merged
+   * but you can save some overhead by only going through the CMap once. Therefore
+   * the extra class.
+   */
+  class OPENMS_DLLAPI ConsensusMapMergerAlgorithm:
+    public DefaultParamHandler,
+    public ProgressLogger
+  {
+  public:
+    ConsensusMapMergerAlgorithm ();
+    /// Takes a cmap with one IDRun per column and merges them to one proteinIDRun per Condition
+    /// while reassociating the PeptideEvidences
+    /// Constructs the mapping based on the exp. design and then uses mergeProteinIDRuns
+    /// @throws MissingInformationException for e.g. missing map_indices in PeptideIDs
+    void mergeProteinsAcrossFractionsAndReplicates(ConsensusMap& cmap, const ExperimentalDesign& exp_design) const;
+
+    /// Similar to above, merges every ID Run into one big run. Proteins get only inserted once but Peptides stay unfiltered
+    /// i.e. might occur in several PeptideIdentifications afterwards
+    /// @throws MissingInformationException for e.g. missing map_indices in PeptideIDs
+    void mergeAllIDRuns(ConsensusMap& cmap) const;
+
+    /// Takes a ConsensusMap and a mapping between ConsensusMap column index (map index) and
+    /// the new ProteinIdentificationRun index and merges them. If you know the number of resulting
+    /// ProteinIDRuns, consider specifying new_size.
+    /// @throws MissingInformationException for e.g. missing map_indices in PeptideIDs
+    void mergeProteinIDRuns(ConsensusMap &cmap,
+                            const std::map<unsigned, unsigned>& mapIdx_to_new_protIDRun) const;
+
+    /// Takes a vector of old protein ID runs and old peptide ID runs, which will be moved or overwritten
+    /// and a map from old run to new run, as well as a to-be-filled vector of peptide IDs
+    /// It merges the proteins from runs that map to the same new run (by moving the first occurence to it)
+    /// It concatenates and moves the peptides of those runs into the vector at the according index while updating their
+    /// run references.
+    void mergeIDRunsAndSplitPeptides(
```

Comment:
purpose or use case?

Reply:
Was my first attempt to do inference on CMaps. Left it just in case. Removed